src/cls/IPM/Main.cls

```diff
@@ -2594,22 +2696,59 @@ ZPM(pArgs...)
  If $System.CLS.IsMthd("%IPM.Main", "Shell") { 
  	Do ##class(%IPM.Main).Shell(pArgs...) Quit
  } Else {
-	// TODO: Decide what to do here; definitely needs to support enabling IPM from here as well,
-	// but need to decide what level of customization to provide
-	Quit ""
+	// TODO: Needs to support enabling IPM from here, also need to decide what level of customization to provide
+	Set tQuitOnError = $Get(pArgs(2))
+	Set tHaltOnComplete = $Get(pArgs(3))
+	Write !, "IPM is not enabled in this namespace."
+	&sql(SELECT %DLIST(Nsp) INTO :namespaces FROM %SYS.Namespace_List())
+	If SQLCODE '= 0 {
+		Write !, "Error getting namespace name. SQLCODE =", SQLCODE
+	} Else {
+		New $Namespace
+		Set ptr = 0
+		Set found = 0	
+		While $ListNext(namespaces, ptr, ns) {
+			Set $Namespace = $Zstrip(ns, "<>WC")
+			If $System.CLS.IsMthd("%IPM.Main", "Shell") {
```

Comment:
I think we need helper method for IsIPMEnabled(). There is an IPMInstalled method below already so we have a single source of truth of such checks (also would rename it to IsIPMInstalled to indicate it returns a boolean).

Reply:
@isc-kiyer the issue is that %IPM may not be mapped everywhere, so we need to duplicate code in the language extension (which is available instance-wide even in places where %IPM is not mapped)